packages/webapp/src/util/task.ts

```diff
@@ -121,9 +121,10 @@ export const formatSoilAmendmentTaskToFormStructure = (
 const formatPurposeIdsToRelationships = (
   purposeIds: number[],
   otherPurpose: string | undefined,
+  otherPurposeId: number,
 ): PurposeRelationship[] => {
   return purposeIds.map((purpose_id) => {
-    return { purpose_id, other_purpose: otherPurpose };
+    return { purpose_id, other_purpose: purpose_id == otherPurposeId ? otherPurpose : undefined };
```

Comment:
@kathyavini 

I am seeing the other_purpose I specify not make it to the database. Is it possible one of these is a string and not the other?